phpBB/adm/style/admin.js

```diff
@@ -261,5 +261,13 @@ function parse_document(container)
 		$('.actions a:has(i.acp-icon)').mouseover(function () {
 			$(this).css("text-decoration", "none");
 		});
+
+		// Live update BBCode font icon preview
+		$('#bbcode_font_icon').on('keyup', function(e) {
+			const iconName = $(this).val();
+			if (iconName.match(/^[\w-]+$/)) {
+				$('#bbcode_icon_preview').attr('class', "o-icon o-icon-font fa-fw fas icon fa-" + $(this).val());
```

Comment:
I wish that all those other classes don't have to be used here, because they could change at any time, and might get missed in an update. Would be better if it could just work on the one font class. 

Like...I don't know...maybe get the class list and look for all that include `fa-` and toggle that with the new font class name. Or loop through the class list, remove any that `startsWith()` `fa-`, then add back to the element the `fa-fw` and whatever the new `fa-icon` name is.

Reply:
You love creating work for me don't you 😂

Reply:
Yep! You can look at what I did in Pages for 4.0 for inspiration. It still relies on naming a bunch of those classes, so I'm not totally happy with it, but I do like it better than what Pages was doing before.
https://github.com/phpbb-extensions/pages/blob/3a44f8f398fd272fe87c6be260fcabc535f15639/adm/style/scripts.js#L45

Reply:
I might have an idea for altering the class name without having to mess with the other class names. The issue with the linked solution is you're still defining the class names in a file so as you say, if they change (like in an FA update), you'll have to remember to update them there.

Perhaps you could get the last class in the list, check it starts with `fa-` but is not `fa-fw` and then remove it and then add the new class name. That should work.

Reply:
Yeah, I'm working on this code right now as an update for pages:
```js
// Remove all classes starting with 'fa-'
Array.from(element.classList).forEach(className => {
	if (className.startsWith('fa-')) {
		element.classList.remove(className);
	}
});

// Add the new class with 'fa-' prefix and 'fa-fw'
element.classList.add('fa-fw', `fa-${newClass}`);
```

Reply:
Just needs to add `and not fa-fw` logic to the conditional

Reply:
Code updated https://github.com/phpbb-extensions/pages/blob/1deee23ef4465d70e6c2a1b9392ef794db6ee20c/adm/style/scripts.js#L45